pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -544,9 +544,7 @@ public void addSegment(ImmutableSegmentImpl segment, @Nullable ThreadSafeMutable
   protected void addOrReplaceSegment(ImmutableSegmentImpl segment, ThreadSafeMutableRoaringBitmap validDocIds,
       @Nullable ThreadSafeMutableRoaringBitmap queryableDocIds, Iterator<RecordInfo> recordInfoIterator,
       @Nullable IndexSegment oldSegment, @Nullable MutableRoaringBitmap validDocIdsForOldSegment) {
-    if (_partialUpsertHandler != null) {
```

Comment:
why this change?

Reply:
Keeping the behaviour same for partial-upsert and full-upsert tables. This is not causing any behavioural changes for full-upsert tables.

Reply:
hmm. but will this be extra cost for full upsert table? as the resolveComparisonTies method iterates records

Reply:
It should be compensated with the update record location part in ConcurrentHashmap flow as say instead of x record-location reference updates per key we will only have 1.

Reply:
Yeah true! Let me know if you want to revert this. 
I was planning to keep the behaviour same for partial-upsert and full-upsert though.

Reply:
sry for the delay. yes, I'd prefer to keep the if-check to resolve ties for partial upsert.

Reply:
Done! I had to move this inside the sub-class as i realised why I added this! For full-upsert, if we don't call `resolveComparisonTies` then we might end-up incrementing the segment-count multiple times but decreasing it only once during `replaceSegment` flow.